rust-toolchain.toml

```diff
@@ -0,0 +1,3 @@
+[toolchain]
+channel = "nightly"
+components = [ "clippy", "miri", "rust-analyzer", "rust-docs", "rust-src", "rustfmt" ]
```

Comment:
Does this have any advantages over using `selectLatestNightlyWith`? Since we're not pinning a specific nightly revision within that file. I suppose it's useful in some way for people not using nix?

Reply:
Yeah, it's read by [rustup](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file).